test/cqlpy/test_vector_similarity.py

```diff
@@ -0,0 +1,176 @@
+            cql.execute(f"SELECT pk, similarity_{similarity_function}('a', 'b') FROM {table}")
+        with pytest.raises(InvalidRequest, match=REQUIRES_FLOAT_VECTOR.format(similarity_function)):
+            cql.execute(f"SELECT pk, similarity_{similarity_function}('a', [1.1, 1.2, 20.25]) FROM {table}")
+        with pytest.raises(InvalidRequest, match=REQUIRES_FLOAT_VECTOR.format(similarity_function)):
+            cql.execute(f"SELECT pk, similarity_{similarity_function}([1.1, 1.2, 20.25], 'b') FROM {table}")
+        with pytest.raises(InvalidRequest, match=REQUIRES_FLOAT_VECTOR.format(similarity_function)):
+            cql.execute(f"SELECT pk, similarity_{similarity_function}({{1.0, 2.0, 3.0}}, {{4.0, 5.0, 6.0}}) FROM {table}")
+        with pytest.raises(InvalidRequest, match=REQUIRES_FLOAT_VECTOR.format(similarity_function)):
+            cql.execute(f"SELECT pk, similarity_{similarity_function}({{1.0, 2.0, 3.0}}, [1.1, 1.2, 20.25]) FROM {table}")
+        with pytest.raises(InvalidRequest, match=REQUIRES_FLOAT_VECTOR.format(similarity_function)):
+            cql.execute(f"SELECT pk, similarity_{similarity_function}([1.1, 1.2, 20.25], {{4.0, 5.0, 6.0}}) FROM {table}")
+
+
+@pytest.mark.parametrize("similarity_function", ["cosine", "euclidean", "dot_product"])
+def test_vector_similarity_returns_null_on_null_arguments(cql, test_keyspace, similarity_function):
+    schema = 'pk int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"INSERT INTO {table} (pk, v) VALUES (1, [1.0, 2.0, 3.0])")
+        result = cql.execute(f"SELECT pk, similarity_{similarity_function}(null, [1.1, 1.2, 20.25]) FROM {table}")
+        for row in result:
+            assert row[1] is None
+        result = cql.execute(f"SELECT pk, similarity_{similarity_function}(v, null) FROM {table}")
+        for row in result:
+            assert row[1] is None
+        with pytest.raises(InvalidRequest, match="Cannot infer type of argument"):
+            cql.execute(f"SELECT pk, similarity_{similarity_function}(null, null) FROM {table}")
+
+
+@pytest.mark.parametrize("similarity_function", ["cosine", "euclidean", "dot_product"])
+def test_vector_similarity_allow_both_vector_columns_and_literals_as_arguments(cql, test_keyspace, similarity_function):
+    schema = 'pk int primary key, v1 vector<float, 3>, v2 vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"INSERT INTO {table} (pk, v1, v2) VALUES (1, [1.0, 2.0, 3.0], [4.0, 5.0, 6.0])")
+        cql.execute(f"SELECT pk, similarity_{similarity_function}([1.1, 1.2, 20.25], [1.8, 0.5, 20.03]) FROM {table}")
+        cql.execute(f"SELECT pk, similarity_{similarity_function}(v1, v2) FROM {table}")
+        cql.execute(f"SELECT pk, similarity_{similarity_function}(v1, [1.8, 0.5, 20.03]) FROM {table}")
+        cql.execute(f"SELECT pk, similarity_{similarity_function}([1.1, 1.2, 20.25], v2) FROM {table}")
```

Comment:
Should also test that the results match expectations (and match Cassandra results for the same queries), so the actual calculations don't regress.